### DISCUSSION TITLE
restrict model output to a defined character set?

### DISCUSSION THREAD

**Discussion by wsyxbcl** [robertknight/ocrs#90] [General]
Is there a way to customize the "dictionary" that the model uses, restricting the output characters to a specified subset? I understand that training a customized model might be the only choice, but I'm curious if there's any workaround.

**Comment by robertknight** [comment]
There isn't a built-in option to customize the output character set at present. If you just want to filter out certain characters you can post-process the outputs of `recognize_text`. If you want to force the model to choose between only an allowed set of characters (for example, always predict numbers), you'd probably want to fine-tune an [existing model](https://huggingface.co/robertknight/ocrs/tree/main) by starting from an existing checkpoint and replacing the final layer. I'm afraid there isn't a guide specifically for fine-tuning yet, but it would be similar to the [training guide](https://github.com/robertknight/ocrs-models/blob/main/docs/training.md).

There is a related issue for supporting other languages/scripts (which could include subsets of the default alphabet) [here](https://github.com/robertknight/ocrs/issues/8).

**Reply by wsyxbcl** [reply]
Thanks for the detailed reply, will try fine-tuning the model and let you know the result.
